src/core/defs.jl

```diff
@@ -578,13 +578,44 @@ unit(obj::ParameterDefReference) = parameter(obj).unit
     variable_dimensions(obj::AbstractCompositeComponentDef, comp_path::ComponentPath, var_name::Symbol)
 
 Return the names of the dimensions of variable `var_name` exposed in the composite 
-component definition indicated by`obj` along the component path `comp_path`.
+component definition indicated by`obj` along the component path `comp_path`. The 
+`comp_path`is of type `Mimi.ComponentPath` with the single arugment being a named
+tuple of symbols describing the relative (to a composite) or absolute (relative to ModelDef) 
+path through composite nodes to specific composite or leaf node.
 """
 function variable_dimensions(obj::AbstractCompositeComponentDef, comp_path::ComponentPath, var_name::Symbol)
     var = variable(obj, comp_path, var_name)
     return dim_names(var)
 end
 
+"""
+    variable_dimensions(obj::AbstractCompositeComponentDef, comp::Symbol, var_name::Symbol)
+
+Return the names of the dimensions of variable `var_name` exposed in the composite 
+component definition indicated by`obj` for the component `comp`, which exists in a
+flat model.
+"""
+function variable_dimensions(obj::AbstractCompositeComponentDef, comp::Symbol, var_name::Symbol)
+    return variable_dimensions(obj, Mimi.ComponentPath((comp,)), var_name)
+end
+
+"""
+    variable_dimensions(obj::AbstractCompositeComponentDef, comp::Symbol, var_name::Symbol)
+
+Return the names of the dimensions of variable `var_name` exposed in the composite 
+component definition indicated by`obj` along the component path `comp_path`. The 
+`comp_path` is a tuple of symbols describing the relative (to a composite) or 
+absolute (relative to ModelDef) path through composite nodes to specific composite or leaf node.
+"""
+function variable_dimensions(obj::AbstractCompositeComponentDef, comp_path::Tuple{N, Symbol}, var_name::Symbol) where N
+    return variable_dimensions(obj, Mimi.ComponentPath((comp_path)), var_name)
+end
+
+# special case for a Tuple with a single element, shouldn't really be used but could come up
+function variable_dimensions(obj::AbstractCompositeComponentDef, comp_path::Tuple{Symbol}, var_name::Symbol) where N
+    return variable_dimensions(obj, Mimi.ComponentPath((comp_path)), var_name)
+end
```

Comment:
Sorry I'm just wrapping my head around this, why do we need the second method? wouldn't a single element tuple stille be handled by the method defined before? If not though, I think the "where N" should be removed

Reply:
I thought so too, but Julia seems to internally simplify the case where there is only one element in the Tuple so it does not get picked up by the first method.  It seems to me unlikely that someone would pass `(comp_name,)` instead of just `comp_name`, but if they did we need to his method for that case. You're right I can remove the `where N`

Reply:
ComponentPath holds an NTuple, not a NamedTuple.

Reply:
The `where N` in the `Tuple{N, Symbol}` allows it to be a Tuple of any number of Symbols.

Reply:
Oh ok, for some reason without the second method it didn't seem to be picking up a call with a single element `Tuple` but I can try again.